packages/bueno/src/schema.ts

```diff
@@ -26,14 +26,23 @@ export class SchemaValidationError extends Error {
   }
 }
 
+function keysOf<T>(obj: T): (keyof T)[] {
+  return Object.keys(obj) as (keyof T)[];
+}
+
+function addDefinedEntries<T>(target: T, source: T): void {
+  keysOf(source).forEach(
+    (key) => source[key] !== undefined && (target[key] = source[key])
+  );
```

Comment:
I'm a little cautious when it comes to magically filtering values, since there might one day be a use-case where this is the way tell an api to do/not do something.

When I see a type written as:

```
currentValues?: string[]
```

I understand this as:

```
The `currentValues` key may or may not be specified. If it is specified, then the value must be an array
```

If `undefined` is passed, then Bueno should throw an error. The value does not follow the typescript type. I don't think we should compensate for this.


Contrast this with:

```
currentValues?: string[] | undefined;
```

which means that if the `currentValues` key is specified, it's value can be `undefined`. In this case, Bueno should not throw an error if the value is `undefined`.

Reply:
Hmm, I assumed `null` was when we meant to specify "nothing" and that `undefined` should be equivalent whether it's `({}).value` or `({ value: undefined }).value`.

If we mean for `undefined` to sometimes be an explicit value, then I'll reject this issue and close this PR.